### ISSUE TITLE
Unable to update 0.90.1 -> 0.91.4

### ISSUE THREAD

**Issue by ofekashery** [home-assistant/hassio#1045]
**Home Assistant release with the issue:**
0.90.1

**Supervisor logs:**
```
19-04-17 08:23:08 INFO (MainThread) [hassio.updater] Fetch update data from https://s3.amazonaws.com/hassio-version/stable.json
19-04-17 08:27:52 ERROR (SyncWorker_17) [hassio.docker.interface] Can't install homeassistant/raspberrypi3-homeassistant:0.91.4 -> 500 Server Error: Internal Server Error ("readlink /mnt/data/docker/overlay2/l: invalid argument").
19-04-17 08:27:52 WARNING (MainThread) [hassio.homeassistant] Update Home Assistant image fails
```

**Comment by pvizeli** [comment]
Corrupt file system :(

**Comment by ofekashery** [comment]
Aww. What can I do? Reinstall the system?

**Comment by dadeboni** [comment]
I have a similar issue going from .91.3 -> .91.4

```
19-04-18 10:52:32 INFO (MainThread) [hassio.utils.gdbus] Call org.freedesktop.systemd1.Manager.ListUnits on /org/freedesktop/systemd1
19-04-18 11:52:54 INFO (MainThread) [hassio.updater] Fetch update data from https://s3.amazonaws.com/hassio-version/stable.json
19-04-18 11:53:05 WARNING (MainThread) [hassio.updater] Can't fetch versions from https://s3.amazonaws.com/hassio-version/stable.json: 
```

**Comment by banana-ben** [comment]
I don’t know of a fix besides reinstalling the system, but if you take a snapshot using the hassio snapshot feature and download the snapshot file somewhere safe (ie not the machine hassio is installed) on you can restore your current hassio setup once you reinstall the system again.

**Comment by Marcoletto** [comment]
Me too, same error.... any ideas to fix?

**Comment by Marcoletto** [comment]
@pvizeli > Corrupt file system :(

In your opinion, why is the system corrupted?

**Comment by dadeboni** [comment]
I fixed mine and was able to update to the latest. My solution was simply to update the reboot Ubuntu server. Once the server was back online and homeassistant was up the ability to update was there. Not sure if that will help anyone but it helped me.
```
sudo apt-get update
sudo apt-get upgrade
sudo reboot
```

**Comment by cogneato** [comment]
@dadeboni unless you were also seeing `"readlink /mnt/data/docker/overlay2/l: invalid argument"` this was not the same issue.

**Comment by VincentMasselis** [comment]
Same issue here when trying to update from 0.92.2 to 0.93.1

```
19-05-19 13:57:22 INFO (SyncWorker_11) [hassio.docker.interface] Pull image homeassistant/raspberrypi3-homeassistant tag 0.93.1.
19-05-19 13:57:23 ERROR (SyncWorker_11) [hassio.docker.interface] Can't install homeassistant/raspberrypi3-homeassistant:0.93.1 -> 500 Server Error: Internal Server Error ("readlink /mnt/data/docker/overlay2/l: invalid argument").
19-05-19 13:57:23 WARNING (MainThread) [hassio.homeassistant] Update Home Assistant image fails
```

**Comment by jnach** [comment]
I installed this image on a rpi 3+ about two weeks ago, and then again today. It worked previously, but must try to update on launch as it's the same image (I tried across two pi's)

I have traditionally used the device-mapper storage driver, vs overlay2, but will report back with findings. It's very clearly a configured option somewhere in the driver settings. Hassio seems to have vi and root access, here goes nothing :)
